include/zephyr/drivers/pwm.h

```diff
@@ -708,14 +706,13 @@ static inline int pwm_configure_capture(const struct device *dev,
 					pwm_capture_callback_handler_t cb,
 					void *user_data)
 {
-	const struct pwm_driver_api *api =
-		(const struct pwm_driver_api *)dev->api;
+	__ASSERT_NO_MSG(DEVICE_API_IS(pwm, dev));
 
-	if (api->configure_capture == NULL) {
+	if (DEVICE_API_GET(pwm, dev)->configure_capture == NULL) {
 		return -ENOSYS;
 	}
 
-	return api->configure_capture(dev, channel, flags, cb,
+	return DEVICE_API_GET(pwm, dev)->configure_capture(dev, channel, flags, cb,
```

Comment:
Instead of doing pointer dereferencing twice, why not leave the local variable? Same for the other, optional API functions.